source/reactor/unit_tests/EventReaderMock.h

```diff
@@ -0,0 +1,29 @@
+#ifndef EventReaderMock_H
+#define EventReaderMock_H
+
+#include "reactor/EventHandlers.h"
+
+#include "gmock/gmock.h"
+
+namespace u {
+namespace reactor {
+class EventLoop;
+
+class EventReaderMock : public EventHandlers {
+
+public:
+    EventReaderMock()
+    {
+        onRead    = [&](int fd, EventLoop *loop) { handleRead(fd, loop); };
+        onError   = [&](int fd, EventLoop *loop) { handleError(fd, loop); };
+        onTimeout = [&](int fd, EventLoop *loop) { handleTimeout(fd, loop); };
+    }
+
+    MOCK_METHOD(void, handleRead, (int fd, EventLoop *loop));
+    MOCK_METHOD(void, handleError, (int fd, EventLoop *loop));
+    MOCK_METHOD(void, handleTimeout, (int fd, EventLoop *loop));
+};
```

Comment:
There are two types of EventHandlers in the tests: one used for reading and one for writing. And we need this differentiation: the reader doesn't define the handleWrite function because otherwise EventLoop would fire an event on each loop iteration.